### DISCUSSION TITLE
Next.js and Vercel cache of fetch

### DISCUSSION THREAD

**Discussion by danielrhodes** [vercel/community#1991] [Help]
### Summary

I have a Next.js 13 project and have been using Vercel to serve the app. The app is dynamic and loads JSON from S3. One thing I have noticed is that even if I pass in a revalidation when using `fetch`, the data that comes back is stale, often times for long periods of time. I have confirmed that the data is up to date on the origin.

An example of the fetch is:

```typescript
fetch(url, {
    next: {
      revalidate: 60,
    }
})
```

When I deploy, the updated app will momentarily show the correct data. But then I just went back right now and refreshed the page and it reverted back to what it was before. If I deploy the app to another service (e.g. Cloudflare Pages), this issue was not there. When I look at the logs for the app in the Vercel dashboard, it says the Cache Status (which I assume to be at the page level) is `MISS`.

Therefore, I suspect that there is an issue in Vercel's underlying caching of `fetch` requests.

I have also put the invalidation of 60s on the page level as well for all relevant pages:

```
export const revalidate = 60;
```

This was also not happening yesterday. Is there anything else I can try? It's hard for me to tell if it's a Vercel caching issue or something I can change in my own code.

**Comment by smaeda-ks** [comment]
@danielrhodes this might be related to: https://github.com/orgs/vercel/discussions/1808#discussioncomment-5373444

**Comment by danielrhodes** [comment]
That could be. I found that if I manipulate the revalidate exports from a page, that seems to change things. However, if I set anything other than `export const revalidate = 0;`, it seems to revert back to a days old version of the cache. My expectation is that this would have expired by now or at the cache would get invalidated on a deploy based on the docs I read.
